GOVERNANCE.md

```diff
@@ -0,0 +1,126 @@
+Governance for the R data.table project
+
+# Purpose and scope
+
+## This document
+
+The purpose of this document is to define how people related to the project work together, so that the project can expand to handle more contributions, from a larger and more diverse group of contributors.
+
+## The R package
+
+The purpose of the project is to maintain the R data.table package,
+which is guided by the following principles:
+
+* Few (if any) dependencies
+* Time & memory efficiency
+* Concise syntax (minimal redundancy in code)
+* Stable code base (easy for users to upgrade to new data.table, and compatible with old R versions)
+* Comprehensive and accessible documentation
+* Clear error and warning messages
+
+To prioritize developer time, we define what is in and out of scope.
+The current scope of package functionality includes:
+* Data manipulation and analysis 
+    * reshaping/pivoting
+    * aggregation/summarizing
+    * subsetting rows
+    * all sorts of joins
+    * adding/updating/deleting columns
+* Reading/writing of data from/to many file formats
+
+Functionality that is out of scope:
+* plotting/graphics (like ggplot2)
+* manipulating data stored on disk (or remote SQL DB) rather than in memory (like sqldf / dbplyr)
+* machine learning / modeling (like mlr3)
```

Comment:
not clear what we mean here -- I think we obviously expect (and encourage!) the package to be used extensively as a tool by ML practitioners. what sorts of features are we ruling out here?

Reply:
For example, I don't think data.table should provide implementation of ML-related data manipulation operations like one-hot encoding

Reply:
Gotcha. I wouldn't totally close off helpful features like this -- `tstrsplit()` and `frollapply()` come to mind. Maybe one-hot encoding is too domain-specific, but if we can offer high-performance (e.g. best-in-class) broad-utility helper functions for data analysis/munging, I don't want to shy away too generically.

Reply:
Pointing out mlr3 next to ggplot2 makes it pretty that we just don't want to duplicate functionality of those great packages. Let me resolve this one by change suggestion below.